### PR TITLE
CockroachDB storage backend

### PR DESCRIPTION
**Pull request by zhaohansprt**
replace the file system to a nice dist db ; uses cgo tested .
it is suitable for server side distrubuted uses ,using db so we can mange all the bot data nicely

### CODE REVIEW

**Comment by neongreen** [comment]
It looks like you've changed line endings and Github thinks that you've changed literally every line of code; please convert line endings back to Unix so that the diff would be easier to review.

**Comment by neongreen** [comment]
Also, could you briefly describe what problems with file-based storage you have stumbled upon that led you to implement CockroachDB support? If they are serious, maybe the official clients should switch to Cockroach too in this case :slightly_smiling_face:

**Comment by zhaohansprt** [comment]
😥 that  is wired  all rust code were written under the ubuntu

**Comment by zhaohansprt** [comment]
it is suitable for server side uses ,uses db so we can mange all the bot data nicely
i am developing a golang version bot sdk for serverside uses

**Comment by neongreen** [comment]
> that is wired all rust code were written under the ubuntu

Try this:

```
git config --global core.autocrlf input
find ./ -type f -exec dos2unix {} \;
```

**Comment by neongreen** [comment]
I see that you're referring to your fork of wireapp/cryptobox -- could you make a PR there as well? Also, it looks like you haven't simply added a Cockroach-based backend, but *replaced* file storage with Cockroach storage. It would be better to have both of them available in cryptobox, because it's unlikely that we will have the time to test CockroachDB properly and switch to it on all clients.

**Comment by zhaohansprt** [comment]
then please make me a new branch for this feature

**Comment by neongreen** [comment]
Sorry, a new branch where / with what exactly?

**Comment by neongreen** [comment]
Also, maybe let's discuss this in chat? https://app.wire.com/join/?key=2Np7Sfj0003BUEu9URtO&code=fbS8_snlnWIgTxROcv-e

**Comment by zhaohansprt** [comment]
it's done PR is ready

File: Cargo.toml [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 
 [lib]
-name         = "cryptobox"
-crate-type   = ["cdylib"]
+name         = "cryptoboxdb"
+crate-type   = ["staticlib","cdylib"]
 
 [dependencies]
-cryptobox = { git = "https://github.com/wireapp/cryptobox", branch = "develop" }
+cryptobox = { git = "https://github.com/zhaohansprt/cryptobox", branch = "crdb" }
```
Comment (neongreen) [inline Cargo.toml]: This should refer to wireapp/cryptobox now (the cockroachdb-experimental branch)

File: Cargo.toml [hunk 2]
```diff
@@ -1,17 +1,21 @@
 [package]
 name         = "cryptobox-c"
-version      = "1.0.0"
+version      = "2.0.0"
```
Comment (neongreen) [inline Cargo.toml]: Perhaps let's not bump the version (unless there's a good reason to?)

File: go/main.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
+package main
```
Comment (neongreen) [inline go/main.go:1]: You've checked `go/cbox` into this branch -- Github says it's a binary. Perhaps that was by accident?

File: Cargo.toml (hunk 1 above)
Comment (zhaohansprt) [reply Cargo.toml]: yes it should be,my overlooked

File: Cargo.toml (hunk 2 above)
Comment (zhaohansprt) [reply Cargo.toml]: ok

File: go/main.go (hunk 3 above)
Comment (zhaohansprt) [reply go/main.go:1]: I uses cgo to test  if this ffi  lib  works well

Comment (neongreen) [reply go/main.go:1]: Sure, but I still don't understand why there's a 2.49 MB binary in the repository :slightly_smiling_face:

Comment (zhaohansprt) [reply go/main.go:1]: i thought you mean main.go this is not a  binay  and cbox is, i have delete now  ,thanks for the check👍

Comment (neongreen) [reply go/main.go:1]: Oh, sorry for the confusion -- I would've commented on `cbox` but Github doesn't let me :unamused:

**Review by neongreen — APPROVED** [review]

File: go/main.go (hunk 3 above)
Comment (zhaohansprt) [reply go/main.go:1]: ok i see😀

**Pull request merged**